myhoard/restore_coordinator.py

```diff
@@ -1135,6 +1135,22 @@ def _check_sql_slave_status(self):
         with self._mysql_cursor() as cursor:
             cursor.execute("SHOW SLAVE STATUS")
             slave_status = cursor.fetchone()
+
+            # Sometimes mysqld can be OOM-killed during backup restoration.
+            # In that case IO and/or SQL thread might not be runnig, so they must be restarted.
+            io_thread_running = slave_status["Slave_IO_Running"]
+            sql_thread_running = slave_status["Slave_SQL_Running"]
+            refetch_slave_status = False
+            if io_thread_running != "Yes":
+                cursor.execute("START SLAVE IO_THREAD")
```

Comment:
IO thread is typically not expected to be running and must not be automatically started.